x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/apikey/ApiKeyRestIT.java

```diff
@@ -176,4 +177,24 @@ public void testGrantApiKeyWithoutApiKeyNameWillFail() throws IOException {
         assertEquals(400, e.getResponse().getStatusLine().getStatusCode());
         assertThat(e.getMessage(), containsString("api key name is required"));
     }
+
+    public void testGrantApiKeyWithoutGrantApiKeyPrivilegeFails() throws IOException {
+        Request request = new Request("POST", "_security/api_key/grant");
+        request.setOptions(
+            RequestOptions.DEFAULT.toBuilder()
+                .addHeader("Authorization", UsernamePasswordToken.basicAuthHeaderValue(END_USER, END_USER_PASSWORD))
+        );
```

Comment:
This does not test the actual change: `END_USER` does not have the `manage_own_api_key` privilege which means the authorization denial does not go through the new changes in `ManageOwnApiKeyClusterPrivilege`.

Reply:
> it might be getting called via findPrivilegesThatGrant, when constructing the exception message.

Good find! So the test actually does excercise the changes but in an unexpected way! I still think it's worth to explicity grant `manage_own_api_key` privilege to the `END_USER`. Then this test can kill both birds with one stone.